Cart

brian7homas opened the pull request:
Base logic for cart feature

brian7homas committed:
feat(Cart Context): SETUP

cart context and reducer added

brian7homas committed:
feat(Cart Context): DISPLAY

displaying cart array length in header

brian7homas committed:
Merging Cart Context setup changes

brian7homas committed:
feat(Cart Context): ADD/REMOVE

added functions to add/remove from cart state using context/reducer

brian7homas committed:
refactor(Cart Context): REMOVE

added logic to remove cart items

brian7homas committed:
fix(Strictmode): STRICTMORE CAUSING ERROR

removed strictmode due to it causing reducer to run twice with 1 click - when adding items to cart
results in duplicate items in cart

gkm0001 commented:
I think you need to fix a lot in the page, you have to think about responsiveness in the page. and products are not visible in to cart option.

brian7homas committed:
feat(Cart): ICON

added cart icon
also setup to import FaBars directly rather than import everything

brian7homas committed:
feat(Cart): ICON

wrapped the cart icon with a div to match the styles of other icons
added state display w/ absolute positioning relative to its container

brian7homas committed:
refactor(Cart): ICON

setup addToCart function
addToCart button only present on the itemInfo page
addToCart pulling item data from localStorage

brian7homas committed:
refactor(ItemCard): ADD TO CART BTN

setup add to cart button w/ conditional

brian7homas committed:
refactor(ItemCard): BUTTONS

added container for buttons and set it to flex and flex wrap

brian7homas committed:
refactor(ItemCard): ADD TO CART BUTTON

reversed to match cart conditional in header
